File: 1911/Solution.java
class Solution {
    public long maxAlternatingSum(int[] nums) {
        long even = 0;
        long odd = 0;
        for (int i = 0; i < nums.length; i++) {
            long tmp1 = Math.max(even, odd + nums[i]);
            long tmp2 = Math.max(odd, even - nums[i]);
            even = tmp1;
            odd = tmp2;
        }
        return even;
    }
}

File: 1911/solution.go
package _1911

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func maxAlternatingSum(nums []int) int64 {
	even := 0
	odd := 0
	for _, num := range nums {
		even, odd = max(even, odd+num), max(odd, even-num)
	}
	return int64(even)
}
